server/athenian/api/openapi/openapi.yaml

```diff
@@ -7320,3 +7377,242 @@ components:
       - from
       - to
       type: object
+    CodeCheckMetricsRequest:
+      additionalProperties: false
+      description: Request for calculating metrics on top of code check runs (CI) data.
+      example:
+        for:
+        - repositories:
+          - github.com/athenianco/athenian-webapp
+          - github.com/athenianco/athenian-api
+          commit_authors:
+          - - github.com/vmarkovtsev
+            - github.com/dpordomingo
+        - repositories:
+          - github.com/athenianco/metadata
+          - github.com/athenianco/athenian-api
+          commit_authors:
+          - - github.com/vmarkovtsev
+            - github.com/se7entyse7en
+        metrics:
+        - chk-suites-count
+        date_from: 2020-01-23
+        date_to: 2021-01-24
+        granularities:
+        - all
+        account: 1
+      properties:
+        for:
+          description: |-
+            Sets of developers and repositories for which to calculate the metrics.
+            The aggregation is `AND` between repositories and developers.
+            The aggregation is `OR` inside both repositories and developers.
+          items:
+            $ref: '#/components/schemas/ForSetCodeChecks'
+          type: array
+        metrics:
+          description: Requested metric identifiers.
+          items:
+            $ref: '#/components/schemas/CodeCheckMetricID'
+          type: array
+        date_from:
+          description: Date from when to start measuring the metrics.
+          format: date
+          type: string
+        date_to:
+          description: Date up to which to measure the metrics.
+          format: date
+          type: string
+        timezone:
+          $ref: '#/components/schemas/TimeZone'
+        granularities:
+          items:
+            $ref: '#/components/schemas/Granularity'
+          type: array
+        quantiles:
+          $ref: '#/components/schemas/Quantiles'
+        account:
+          description: Session account ID.
+          type: integer
+        split_by_check_runs:
+          description: Calculate metrics separately for each number of check runs in suite.
+          type: boolean
+      required:
+      - account
+      - date_from
```

Comment:
date_from and date_to define an interval. A set of checks belongs to that interval according to their starting date. However, for the checks that map to PRs, we need to go backward and include all the checks run into those PRs to be able to compute accurately metrics such as:
- suites_per_pr
- suite_time_per_pr
- prs_with_checks_count